src/bsoncxx/types.hpp

```diff
@@ -309,6 +314,11 @@ struct BSONCXX_API b_date {
         : value(std::chrono::duration_cast<std::chrono::milliseconds>(tp.time_since_epoch())) {
     }
 
+    ///
+    /// Default constructor for b_date.
+    ///
+    BSONCXX_INLINE b_date() = default;
+
     std::chrono::milliseconds value;
```

Comment:
I might go a little safer on this one and zero the value here, rather than leaving it uninitialized memory.